File: array/1385_find_the_distance_value_between_two_arrays/1385_find_the_distance_value_between_two_arrays.py

class Solution(object):
    def findTheDistanceValue(self, arr1, arr2, d):
        ans = 0
        for n1 in arr1:
            if all(abs(n1-n2) > d for n2 in arr2):
                ans+=1
        return ans


arr1= [4,5,8]
arr2 = [10,9,1,8]
d = 2
res = Solution().findTheDistanceValue(arr1, arr2, d)
print(res)

File: array/1385_find_the_distance_value_between_two_arrays/1385_find_the_distance_value_between_two_arrays.go
package main

import "fmt"

func findTheDistanceValue(arr1 []int, arr2 []int, d int) int {
	var (
		i, j, rst int
		length1   = len(arr1)
		length2   = len(arr2)
	)
	for i = 0; i < length1; i++ {
		fail := true
		for j = 0; j < length2; j++ {
			if abs(arr1[i]-arr2[j]) <= d {
				fail = false
			}
		}
		if fail {
			rst++
		}
	}
	return rst
}

func abs(a int) int {
	if a < 0 {
		return -a
	}
	return a
}

func main() {
	arr1 := []int{4, 5, 8}
	arr2 := []int{10, 9, 1, 8}
	d := 2
	res := findTheDistanceValue(arr1, arr2, d)
	fmt.Println(res)
}
